sched/scheduler/stateful_scheduler.go

```diff
@@ -437,7 +439,24 @@ func generateJobId() string {
 func (s *statefulScheduler) loop() {
 	for {
 		s.step()
-		time.Sleep(TickRate)
+
+		// wait until our TickRate has elapsed or we have a pending action.
+		// resend messages asynchronously in case the channel is blocked.
+		select {
+		case msg := <-s.checkJobCh:
+			go func() {
+				s.checkJobCh <- msg
+			}()
+		case msg := <-s.addJobCh:
+			go func() {
+				s.addJobCh <- msg
```

Comment:
If we have multiple addJob requests, won't this reorder the addJobCh? Could a job be stuck at the bottom of the addJobCh queue if we keep receiving new requests (>1 every TickRate)?

Reply:
Yes it'll reorder, but only jobs received in the last tickrate window, which won't be perceptible. We won't starve anything though, once we have any jobs to add, the addJob function will drain the channels until they are empty within one step.